### PR TITLE
questions crud

### PR DESCRIPTION
**Pull request by agarwalrashika00**

### CODE REVIEW

File: app/models/question.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  def self.ransackable_attributes(auth_object = nil)
+    ['active', 'created_at', 'description', 'id', 'score', 'title', 'updated_at']
+  end
+
+  def self.ransackable_associations(auth_object = nil)
+    []
+  end
+
+end
```
Comment (Ankita9818) [inline app/models/question.rb:22]: New line

File: app/models/question.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class Question < ApplicationRecord
+
+  include Models::Activable
+  include Sluggable
+
+  validates :title, presence: true
+  validates :slug, presence: true, uniqueness: true
+  validates :score, numericality: true
```
Comment (Ankita9818) [inline app/models/question.rb:8]: Can score be negetive??

File: app/models/question.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Question < ApplicationRecord
+
+  include Models::Activable
+  include Sluggable
+
+  validates :title, presence: true
```
Comment (Ankita9818) [inline app/models/question.rb:6]: Title should be unique too

File: app/views/admin/questions/_question.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<tr class="h-50 body-row">
+  <td><%= question.title %></td>
+  <td><%= question.description %></td>
+  <td><%= question.score %></td>
+  <td>
+    <%= link_to '&#x270D;'.html_safe, edit_admin_question_path(question), class: 'btn btn-info', title: 'edit' %>
```
Comment (Ankita9818) [inline app/views/admin/questions/_question.html.erb:6]: Start using icons

File: config/routes.rb [hunk 5]
```diff
@@ -24,6 +24,11 @@
       patch 'activate', on: :member
       patch 'inactivate', on: :member
     end
+
+    resources :questions, param: :slug do
+      patch 'activate', on: :member
```
Comment (Ankita9818) [inline config/routes.rb:29]:
```
       member do
        patch :activate
        patch :inactivate
      end
```

File: db/migrate/20230526062908_create_questions.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+class CreateQuestions < ActiveRecord::Migration[7.0]
+  def change
+    create_table :questions do |t|
+      t.string :slug, null: false
+      t.string :title
```
Comment (Ankita9818) [inline db/migrate/20230526062908_create_questions.rb:5]: add null false contraint

**Review by Ankita9818 — CHANGES_REQUESTED** [review]

File: app/views/admin/questions/_question.html.erb (hunk 4 above)
Comment (agarwalrashika00) [reply app/views/admin/questions/_question.html.erb:6]: https://trello.com/c/lUzuPSSH/17-icons-instead-of-unicode

**Review by Ankita9818 — APPROVED** [review]

**Pull request merged**